orchestra/models/core/mixins.py

```diff
@@ -174,6 +174,23 @@ def is_entry_level(self):
     def is_reviewer(self):
         return self.assignment_counter > 0
 
+    def _can_view_tag(self, tag, certifications):
+        for tag_certification in tag.get('certificates', []):
+            for worker_certification in certifications:
+                cert = worker_certification.certification
+                if (cert.slug == tag_certification['slug'] and
+                        cert.workflow.slug == tag_certification['workflow']):
+                    return True
+        return False
+
+    def get_assignment_tags(self):
+        tags = self.task.project.project_data.get('tags', [])
```

Comment:
project_data has no implied schema, so we shouldn't impose/assume one. the solution sketch should look as follows
* Add a field called `task_tags_function` to `Step` (similarly could be done on an assignment). This would be similar to `execution_function` (https://github.com/b12io/orchestra/blob/master/orchestra/models/core/models.py#L142)
* Dynamically load the function and run it on the task (or assignment)
* If you're interested in reading `project_data` (or any other approach), you can do that in a specific `task_tags_function`

Alternatively, given that above I'm saying to move this logic onto tasks from assignments, add a field called `tags` to `Task` (requires a schema migration), specify the JSON schema of that field, and read directly from there instead of running any functions.